slicerator/__init__.py

```diff
@@ -461,9 +508,24 @@ def pipeline(func=None, **kwargs):
     When the function is applied to any other object, it falls back on its
     normal behavior.
 
+    Parameters
+    ----------
+    func : callable or type
+        Function or class type for lazy evaluation
+    retain_doc : bool, optional
+        If True, don't modify `func`'s doc string to say that it has been
+        made lazy. Defaults to False
+    argc : int or 'all', optional
+        Number of arguments that are relevant for the pipeline. For instance,
```

Comment:
A bit unclear to me, may be something like `'Number of arguments that are indexed by the pipeline'`, and the name itself, `argc`, could be turned into the more comprehensive `indexed_arg_count`? Or `ancestor_count`?

Reply:
I fixed the doc and went with ancestor_count to keep it reasonably short.